Client/Frontend/Browser/OpenWithSettingsViewController.swift

```diff
@@ -0,0 +1,104 @@
+        return plistSchemes
+    }()
+
+    private let prefs: Prefs
+    private var currentChoice: String!
+
+    private let BasicCheckmarkCell = "BasicCheckmarkCell"
+
+    init(prefs: Prefs) {
+        self.prefs = prefs
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = Strings.SettingsOpenWithSectionName
+
+        tableView.accessibilityIdentifier = "OpenWithPage.Setting.Options"
+
+        tableView.registerClass(UITableViewCell.self, forCellReuseIdentifier: BasicCheckmarkCell)
+        tableView.backgroundColor = UIConstants.TableViewHeaderBackgroundColor
+
+        let headerFooterFrame = CGRect(origin: CGPointZero, size: CGSize(width: self.view.frame.width, height: UIConstants.TableViewHeaderFooterHeight))
+        let headerView = SettingsTableSectionHeaderFooterView(frame: headerFooterFrame)
+        headerView.titleLabel.text = Strings.SettingsOpenWithPageTitle
+        headerView.showTopBorder = false
+        headerView.showBottomBorder = true
+
+        let footerView = SettingsTableSectionHeaderFooterView(frame: headerFooterFrame)
+        footerView.showTopBorder = true
+        footerView.showBottomBorder = false
+
+        tableView.tableHeaderView = headerView
+        tableView.tableFooterView = footerView
+    }
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        self.currentChoice = self.prefs.stringForKey("MailToOption") ?? mailSchemes[0]["scheme"]
+        tableView.reloadData()
+    }
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        self.prefs.setString(currentChoice, forKey: "MailToOption")
+    }
+
+    internal func loadMailSchemes() -> [[String:String]] {
+        if let path = NSBundle.mainBundle().pathForResource("MailSchemes", ofType: "plist"), let dictRoot = NSArray(contentsOfFile: path) {
+            var schemes: [[String:String]] = []
+            dictRoot.forEach({ dict in
+                let dictionary = dict as! [String:String]
+                schemes.append(dictionary)
+            })
+            return schemes
+        }
+        return []
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(BasicCheckmarkCell, forIndexPath: indexPath)
+
+        let option = mailSchemes[indexPath.row]
+
+        guard let schemeName = option["name"], let scheme = option["scheme"], let schemeURL = NSURL(string: scheme) else {
+            return cell
+        }
+
+        cell.textLabel?.attributedText = NSAttributedString.tableRowTitle(schemeName)
+
+        let enabled = UIApplication.sharedApplication().canOpenURL(schemeURL) ?? false
```

Comment:
I would try to avoid calling `canOpenURL` here. If we start adding to this list, the cells would get recycled and we could get a lot of calls. I would cache this value and calculate it in viewWillAppear so we're fresh in the case a user comes back to the screen. Also, we'd probably want to invalidate the cache when we come back from background so it might be useful to add an observer for the `DidEnterForeground` notification just in the case the user backs out and installs the mail app.